src/modules/availability/availability.service.ts

```diff
@@ -0,0 +1,120 @@
+import * as moment from 'moment-timezone';
+import {
+  BadRequestException,
+  ConflictException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Availability } from '@entities/Availability';
+import { Repository } from 'typeorm';
+import { CreateAvailabilityDto } from './dto/create-availability.dto';
+
+@Injectable()
+export class AvailabilityService {
+  constructor(
+    @InjectRepository(Availability) private repo: Repository<Availability>,
+  ) {}
+
+  async createMultiples(
+    dto: CreateAvailabilityDto[],
+    doctorId: number,
+  ): Promise<Availability[]> {
+    if (!Array.isArray(dto)) {
+      throw new BadRequestException('Expected many Availabilities');
+    }
+
+    const savedAvailabilities = await Promise.all(
+      dto.map(async (availability) => {
+        const existingAvailability = await this.repo
+          .createQueryBuilder('availability')
+          .leftJoinAndSelect('availability.doctor', 'doctor')
+          .andWhere('doctor.id = :doctorId', { doctorId })
+          .andWhere('availability.startTime <= :endTime', {
+            endTime: availability.endTime,
+          })
+          .andWhere('availability.endTime >= :startTime', {
+            startTime: availability.startTime,
+          })
+          .getMany();
+        if (existingAvailability.length > 0) {
```

Comment:
I don't think we should use approach to map requests to DB, better set needed columns to unique. Or select all data once and check it.